### ISSUE TITLE
Unblock request for 38.103.165.28

### ISSUE THREAD

**Issue by foozmeat** [CIDRAM/CIDRAM#164]
## Issue
Our public IP is 38.103.165.28 which falls into the Cogent blocklist. We've had this IP for a very long time and are a legitimate business https://panic.com

We own `38.103.165.0/28` and `38.103.165.32/27` in total but really just the one IP needs to be whitelisted.

Thanks

**Comment by DanielRuf** [comment]
Hi @foozmeat,

Can you explain why your IP would appear as visitor of websites? Because CIDRAM blocks only non-user IPs.

**Comment by DanielRuf** [comment]
Or does this prevent you and the users in your network from visiting CIDRAM protected websites?

I try to understand what the current problem is in detail so we know it's not just about listing the IP but also about resolving an actual issue caused by this.

**Comment by foozmeat** [comment]
One of my users is trying to view http://www.myaudia3etron.com and is getting the following error
![image](https://user-images.githubusercontent.com/419456/72299228-0be52080-3615-11ea-9f30-e86d5c70a3d8.png)

**Comment by DanielRuf** [comment]
Ok, thanks for the clarification.

**Comment by Maikuolan** [comment]
Request looks good. :+1:

About to rush out the door for work, so won't have time to do much for a few hours, but I'll figure out the best way to proceed with this and will reply in more detail as soon as I get back. :-)

**Comment by Maikuolan** [comment]
So.. /8s are always a pain to work with (in the context of CIDR-based blocklists and in avoiding false positives), as is any range that size or greater and thereabouts (would generally prefer to not block them at all, if at all avoidable, TBH), and in this particular case, the block event occurred because, in CIDRAM's default signature files (specifically, the `ipv4_isps.dat` signature file), the PSINet signatures include the range `38.0.0.0/8`.

Because PSINet contains a whole pile of just about everything (honest, good, legit human users, numerous clean subordinate ranges, as well as numerous, extremely toxic ranges, numerous websites, hosting services, along with extremely numerous spammers, with huge numbers of spam reports at SFS and elsewhere and so on), but announces everything as just "PSINet", and in addition, includes that /8 (plus, many of the specific IPs that the spammers in question are coming from don't have any smaller subordinate range than that /8, which makes splitting them out a little tedious), simply doing a copy/paste job from BGP records as a basis for updating that signature section isn't exactly ideal, but it what has been happening for a while. To make bypasses easier, I did have a specially crafted rule, specifically for that signature, to avoid triggering multiple signatures in cases where signatures from other sections had already triggered under that particular /8.

All that considered, outright whitelisting or subtracting out the particular ranges, which have been requested here to be unblocked, from the aggregate whenever updating that signature section might've sounded like the most straightforward, simplest solution, but my concern was the high probability of seeing other, similar requests from others in the future, relating to the exact same /8.

I also happened to notice that `38.103.0.0/16` (a subordinate of `38.0.0.0/8`, and the particular /16 containing both the specific ranges requested to be unblocked, `38.103.165.0/28` and `38.103.165.32/27`) seems to (AFAICT) be completely clean at the moment at all the usual databases/services/etc that I normally check against when researching networks/ranges/etc. So I thought, why not just go through and check which other /16s from that /8 happen to seem to be completely clean at the moment, subtract them from the final aggregate, and set that as guideline to use for when updating that particular signature section?

That's what I've done here just now. Still not exactly ideal, and merely a band-aid solution for the mess that is `38.0.0.0/8`, but at least, for the moment, it means that the particular requested ranges are completely removed from CIDRAM's default signature files, and potentially might avoid the few odd similar removal requests here and there that we might end up seeing here in the future.

(Pertinent commits auto-linked by GitHub to this issue, just above this particular reply).

I've gone ahead and marked this issue as "fixed", seeing as, strictly speaking, the request to unblock those ranges has now been fulfilled.

Though, before I close the issue: Anyone have any other thoughts about that /8, about my "band-aid solution", ideas for possible alternative solutions, etc..?
